### ISSUE TITLE
Could not finish Test Step - "Test Name" tempId not found

### ISSUE THREAD

**Issue by NaitikPandya** [reportportal/agent-js-jest#153]
I have a react project that uses the jest for test cases.

So i have done the configuration as per your documentation.
Here is my jest configurations.
```json
"jest": {
    "preset": "ts-jest",
    "testTimeout": 100000,
    "maxConcurrency": 10,
    "maxWorkers": "50%",
    "transform": {
      "^.+\\.(ts|tsx)?$": "ts-jest",
      "^.+\\.(js|jsx)$": "babel-jest",
      "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/fileTransformer.js"
    },
    "moduleNameMapper": {
      "\\.(css|less)$": "identity-obj-proxy"
    },
    "moduleDirectories": [
      "node_modules",
      "src"
    ],
    "testPathIgnorePatterns": [
      "<rootDir>/node_modules/"
    ],
    "setupFiles": [
      "jest-canvas-mock",
      "whatwg-fetch",
      "./src/setupJest.ts"
    ],
    "reporters": [
      "default",
      [
        "@reportportal/agent-js-jest",
        {
          "api_key": "<My API KEY>"
          "endpoint": "http://<URL>:8080/api/v1",
          "project": "test_results",
          "description": "Test result",
          "launch": "demo_test",
          "isLaunchMergeRequired": true,
          "launchUuidPrint": true,
          "restClientConfig": {
            "timeout": 0
          }
        }
      ]
    ]
  }
```
Commonad i'm using to run test cases is
`jest --env=jsdom --no-colors -i`

still i'm getting this error
`Could not finish Test Step - "Test Name" tempId not found`

**Screenshot of reported test in to reportportal**
<img width="1378" alt="Screenshot 2024-11-07 at 4 16 57 PM" src="https://github.com/user-attachments/assets/0d569974-d630-4b84-bc87-b3e4b121fe48">
*I Have more the 600+ test cases but only this much is getting log into the report portal*

Does any one have any idea so please guide through this as i'm not able getting any proper document & same issue to look and resolve it.

It would be great help if anyone reply as soon as possible.

**Comment by owens-ben** [comment]
I'm having the same issue where the same suite is getting broken up into separate ones, like in the screenshot in the description.

![image](https://github.com/user-attachments/assets/dec147cb-50eb-4d9f-a1df-6d2f5aebee2b)

Version 5.1.1

**Comment by tristanzander** [comment]
Are you using anything like `jest.concurrent` or the like? If you can reproduce the issue as a regression test in this repo, I may be able to fix it when I have a moment

**Comment by owens-ben** [comment]
@tristanzander Yes - I'm using Detox which allows multiple executions with --maxWorkers. The issue is new since upgrading from 5.0.6->5.1.1.

What happens is when running concurrently, Jest will frequently invoke _finishSuite prematurely, thus causing it to start a new suite.

**Comment by nishant-shah-social** [comment]
I am also facing the same issue
We are running the jest tests with below flags:
`--maxWorkers=12 --bail=5 --ci --verbose --silent --noStackTrace --coverage --testTimeout=60000 --forceExit`
In the logs I see below lines
```
Could not finish Test Step - "<test> - <scenario name>. tempId not found

Could not finish Test Step - "<test> - <scenario name>. tempId not found
```
In the report portal I see 2 records for the same execution and only 1 test being reported instead of 10

**Comment by rekapallym** [comment]
I am facing the same issue, debugging into the repo looks like `onTestCaseStart` is not getting called by jest, that were the startStep method is called to set the tempId. can anyone check? i am using jest 27.2.5

**Comment by nishant-shah-social** [comment]
I downgraded my agent version to `"@reportportal/agent-js-jest": "5.0.8"`
Now, its working without any issues
Not sure whats wrong with the latest version

**Comment by AmsterGet** [comment]
Hello guys!

Try the latest version [5.1.2](https://github.com/reportportal/agent-js-jest/releases/tag/v5.1.2), which is supposed to fix the mentioned suites issues.
Please let me know if everything now works as expected.
I look forward to your feedback!

**Comment by naryniecki** [comment]
5.1.2 does not solve the issue for me, only downgrading to 5.0.8 worked.

**Comment by AmsterGet** [comment]
Hello @naryniecki !
Thanks for your feedback.

Do you see the same errors?

**Comment by naryniecki** [comment]
Hi, @AmsterGet,
yes, getting:
Could not finish Test Step - "...". tempId not found

**Comment by AmsterGet** [comment]
Hello guys!

Please take a look at [v5.1.3](https://github.com/reportportal/agent-js-jest/releases/tag/v5.1.3).
I found and fixed a test name collision that could potentially cause the problem.
Unfortunately, I still can't reproduce the original error in different setups.
So please check out the fix and let me know if it helps!

**Comment by AmsterGet** [comment]
Hello @naryniecki !
Have you been able to test mentioned version?
Looking forward to your feedback!
